h/services/flag.py

```diff
@@ -7,6 +7,25 @@ class FlagService:
     def __init__(self, session):
         self._session = session
 
+        self._flagged_cache = {}
+
+    def create(self, user: User, annotation: Annotation):
+        """
+        Create a flag for the given user and annotation.
+
+        We enforce the uniqueness of a flag, meaning one user can only
+        flag one annotation once. This method first checks if the annotation
+        is already flagged by the user, if that is the case, then this
+        is a no-op.
+
+        :param user: The user flagging the annotation.
+        :param annotation: The annotation to be flagged.
+        """
+        if self.flagged(user, annotation):
+            return
+
+        self._session.add(Flag(user=user, annotation=annotation))
+
```

Comment:
I moved this up because it seemed weird alphabetically to have it between the two types of flag this, flag that